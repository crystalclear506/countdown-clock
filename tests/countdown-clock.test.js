import { expect, test } from '@jest/globals';
import React from 'react';
import renderer from 'react-test-renderer';
import CountdownClock from '../src/Components/countdown-clock';

jest.useFakeTimers();

test('Component should be rendered correctly', () => {
    const testRenderer = renderer.create(
        <CountdownClock/>
    );
    
    expect(testRenderer).toMatchSnapshot();
});

test('Countdown should decrement correctly', (done) => {
    const testRenderer = renderer.create(
        <CountdownClock/>
    );

    const component = testRenderer.getInstance();

    const seconds = 1000 * 60;
    const timeLater = new Date().getTime() + seconds;
    component.setCountdownEnd(timeLater);
    
    
    setTimeout(() => {
        expect(component.state.secondsDisp).toBe(30);
    }, seconds / 2);

    setTimeout(() => {
        expect(component.state.secondsDisp).toBe(0);
    }, seconds);    
    
    jest.runAllTimers();
    done();
})