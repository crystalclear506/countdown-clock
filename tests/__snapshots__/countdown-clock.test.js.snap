// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should be rendered correctly 1`] = `
<div
  className="clock-container"
>
  <div
    className="clock"
  >
    <div
      className="cell"
    >
      <div>
        0
      </div>
      <div
        className="label"
      >
         Days 
      </div>
    </div>
    <div
      className="cell"
    >
      <div>
        00
      </div>
      <div
        className="label"
      >
         Hours 
      </div>
    </div>
    <div
      className="cell"
    >
      <div>
        00
      </div>
      <div
        className="label"
      >
         Minutes 
      </div>
    </div>
    <div
      className="cell"
    >
      <div>
        00
      </div>
      <div
        className="label"
      >
         Seconds 
      </div>
    </div>
  </div>
</div>
`;
